src/yuzu/game_list_worker.cpp

```diff
@@ -336,18 +336,45 @@ void GameListWorker::ScanFileSystem(ScanTarget target, const std::string& dir_pa
                     }
                 }
             } else {
-                std::vector<u8> icon;
-                [[maybe_unused]] const auto res1 = loader->ReadIcon(icon);
+                std::vector<u64> program_ids;
+                loader->ReadProgramIds(program_ids);
+
+                if (res2 == Loader::ResultStatus::Success && program_ids.size() > 1 &&
+                    (file_type == Loader::FileType::XCI || file_type == Loader::FileType::NSP)) {
+                    for (const auto program_id : program_ids) {
```

Comment:
Otherwise this will conflict with the declaration of `program_id` a couple lines above
```suggestion
                    for (const auto id : program_ids) {
```